internal/configs/annotations.go

```diff
@@ -275,13 +283,24 @@ func parseAnnotations(ingEx *IngressEx, baseCfgParams *ConfigParams, isPlus bool
 		}
 	}
 
-	ports, sslPorts := getServicesPorts(ingEx)
-	if len(ports) > 0 {
-		cfgParams.Ports = ports
+	if values, exists := ingEx.Ingress.Annotations["nginx.org/listen-ports"]; exists {
```

Comment:
The functionality in `getServicesPorts()` was moved in to `ParsePortList()`, so that it can be reused in `validation.go`.